emcc.py

```diff
@@ -983,8 +983,9 @@ def check(input_file):
         shared.Settings.RELOCATABLE = 1
         shared.Settings.PRECISE_I64_MATH = 1 # other might use precise math, we need to be able to print it
         assert not options.use_closure_compiler, 'cannot use closure compiler on shared modules'
-        # getMemory is used during startup of shared modules (to allocate their static memory)
+        # shared modules need memory utilities to allocate their memory
         shared.Settings.EXPORTED_RUNTIME_METHODS += [
+          'allocate',
```

Comment:
Looks good but I don't understand why this is suddenly required, as other changes removed `allocate()` call, not added.

Reply:
(Somehow Closure removed it?)

Reply:
This is for shared modules - allocate is necessary in side modules, so the main module must keep it around. it used to remain alive because we weren't able to remove it, but now we are, so we must keep it alive manually in main modules.